spec/Section 6 -- Execution.md

```diff
@@ -666,6 +665,23 @@ CompleteValue(fieldType, fields, result, variableValues):
     * Let {subSelectionSet} be the result of calling {MergeSelectionSets(fields)}.
     * Return the result of evaluating {ExecuteSelectionSet(subSelectionSet, objectType, result, variableValues)} *normally* (allowing for parallelization).
 
+**Coercing Internal Values**
+
+The primary purpose of value completion is to ensure that the internal values
+returned by field resolvers are valid according to the GraphQL type system and
+a service's schema. This "dynamic type checking" allows GraphQL to provide
+consistent guarantees about returned types atop any service's internal runtime.
+
+See the Scalars [Result Coercion and Serialization](#sec-Scalars.Result-Coercion-and-Serialization)
+sub-section for more detailed information about how GraphQL's built-in scalars
+coerce internal values to GraphQL values.
+
+CoerceInternalValue(leafType, internalValue):
+  * Return the result of calling the internal method provided by the type
+    system for determining the "result coercion" of {leafType} given the value
+    {internalValue}. This internal method must return a valid value for the
+    type or otherwise throw a field error.
```

Comment:
I would add that "result coercion" can't be null, otherwise, we are breaking the symmetry of coercing values and completing values. It can happen in the following scenario we have scalar with `parseValue`/`serialize`(graphql-js terminology) so `serialize` is return null, but if you pass `null` as the input value of the same scalar it will be interpreted by input coercing algorithm and will never hit `parseValue`.

Reply:
Great feedback